neurovault/apps/statmaps/views.py

```diff
@@ -447,51 +444,40 @@ def add_image_for_neurosynth(request):
         priv_token = generate_url_token()
         temp_collection = Collection(name=temp_collection_name,
                                      owner=request.user,
-                                     private=False,
+                                     private=is_private,
                                      private_token=priv_token)
         temp_collection.save()
     image = StatisticMap(collection=temp_collection)
+    redirect = redirect_url+"/?neurovault=%s-%s" % (
+        temp_collection.private_token,image.id)
```

Comment:
Ah I see! Could you take it outside of the function? So the string passed on would be "http://neurosynth.org/decode/?neurovault=%s-%s" instead of "http://neurosynth.org/decode/". It will give more flexibility for future apps using this functionality.